### DISCUSSION TITLE
How do I return only 1 hit with blastx

### DISCUSSION THREAD

**Discussion by sr320** [RobertsLab/resources#1086] [Q&A] [answered]
Here is the code

```
!{bldir}blastx \
-query ../data/GCF_000297895.1_oyster_v9_cds_from_genomic.fna \
-db ../blastdb/uniprot_sprot_2020_06  \
-out ../analyses/Cg-uniprot_blastx.tab \
-evalue 1E-20 \
-num_threads 3 \
-max_target_seqs 1 \
-outfmt 6
```

and for a given query I get multiple hits. Is there a way to reduce to only one hit either in the code or post blast?

**Comment by kubu4** [comment, marked as answer]
I think if you add `-max_hsps 1` as an argument, you should get what you want.

**Comment by sr320** [comment]
And I believe after the fact in R, this works

```
distinct(uniprot, V1, .keep_all = TRUE)
```

**Comment by kubu4** [comment]
Also, post-BLAST, you can use this:

`awk '!array[$1]++' outfmt6.file >tophits.txt`
